spec/support/command_helpers.rb

```diff
@@ -502,22 +502,22 @@ def switch_input indx
 
   def switch_detector_logic
     indx = 0
-    component = Validator.config['components']['detector_logic'].keys[indx]
-
-    force_detector_logic component, mode:'True'
-    Validator.config['main_component'] = Validator.config['main_component']
-    wait_for_status(@task,
-      "activate detector logic #{component}",
-      [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}1/}]
-    )
-    
-
-    force_detector_logic component, mode:'False'
-    Validator.config['main_component'] = Validator.config['main_component']
-    wait_for_status(@task,
-      "deactivate detector logic #{component}",
-      [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}0/}]
-    )
+    Validator.config['components']['detector_logic'].keys.each do |component|
+      force_detector_logic component, mode:'True'
+      Validator.config['main_component'] = Validator.config['main_component']
+      wait_for_status(@task,
+        "activate detector logic #{component}",
+        [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}1/}]
+      )
+      
+      force_detector_logic component, mode:'False'
+      Validator.config['main_component'] = Validator.config['main_component']
+      wait_for_status(@task,
+        "deactivate detector logic #{component}",
+        [{'sCI'=>'S0002','n'=>'detectorlogicstatus','s'=>/^.{#{indx}}0/}]
+      )
+      indx = indx + 1
```

Comment:
you can use each_with_index, then you don't have to manually count

Reply:
Should be fixed now